public/app/features/dashboard-scene/panel-edit/VizPanelManager.tsx

```diff
@@ -13,38 +13,41 @@ import {
 } from '@grafana/data';
 import { config, getDataSourceSrv, locationService } from '@grafana/runtime';
 import {
-  SceneObjectState,
-  VizPanel,
-  SceneObjectBase,
-  SceneComponentProps,
-  sceneUtils,
   DeepPartial,
-  SceneQueryRunner,
-  sceneGraph,
-  SceneDataTransformer,
   PanelBuilders,
+  SceneComponentProps,
+  SceneDataTransformer,
   SceneGridItem,
+  SceneObjectBase,
   SceneObjectRef,
+  SceneObjectState,
+  SceneQueryRunner,
+  VizPanel,
+  sceneGraph,
+  sceneUtils,
 } from '@grafana/scenes';
 import { DataQuery, DataTransformerConfig, Panel } from '@grafana/schema';
 import { useStyles2 } from '@grafana/ui';
 import { getPluginVersion } from 'app/features/dashboard/state/PanelModel';
 import { getLastUsedDatasourceFromStorage } from 'app/features/dashboard/utils/dashboard';
 import { storeLastUsedDataSourceInLocalStorage } from 'app/features/datasources/components/picker/utils';
+import { updateLibraryVizPanel } from 'app/features/library-panels/state/api';
 import { updateQueries } from 'app/features/query/state/updateQueries';
 import { GrafanaQuery } from 'app/plugins/datasource/grafana/types';
 import { QueryGroupOptions } from 'app/types';
 
+import { LibraryVizPanel } from '../scene/LibraryVizPanel';
 import { PanelRepeaterGridItem, RepeatDirection } from '../scene/PanelRepeaterGridItem';
 import { PanelTimeRange, PanelTimeRangeState } from '../scene/PanelTimeRange';
 import { gridItemToPanel } from '../serialization/transformSceneToSaveModel';
-import { getDashboardSceneFor, getPanelIdForVizPanel, getQueryRunnerFor } from '../utils/utils';
+import { getDashboardSceneFor, getLibraryPanel, getPanelIdForVizPanel, getQueryRunnerFor } from '../utils/utils';
 
 export interface VizPanelManagerState extends SceneObjectState {
   panel: VizPanel;
   sourcePanel: SceneObjectRef<VizPanel>;
   datasource?: DataSourceApi;
   dsSettings?: DataSourceInstanceSettings;
+  libraryPanel?: LibraryVizPanel;
```

Comment:
Hmm, good catch. This was implemented before we had the sourcePanel in the state, and we needed the libraryPanel because the panel was disconnected from it's parent. But now that we have the source panel ref we can use the parent.